### DISCUSSION TITLE
Is there high level overview/documentation for how pyright works?

### DISCUSSION THREAD

**Discussion by tjgurwara99** [microsoft/pyright#8471] [Q&A] [answered]
I believe it would be very beneficial for newcomers (like myself) to get familiar with the project in case they are considering contributing to it or learning from it.

**Comment by erictraut** [comment, marked as answer]
You can find a high-level overview [here](https://microsoft.github.io/pyright/#/internals). There are also header comments in each module that explain the purpose of the code in that module.

**Reply by tjgurwara99** [reply]
Great, thanks a lot! I'll give that some reading. The binding module is the main interesting one from my perspective, given that you've written it, could you recommend some resources (books/videos) that would help me understand that concepts used in that module better?

**Reply by erictraut** [reply]
The basic terminology ("symbols", "scopes", "descriptors", "parse trees", etc.) are pretty standard in compiler designs, so if you're not familiar with compiler basics, I recommend starting with text books on that topic.
